### PR TITLE
London Class 8 - Marina Almeida - JS-Core2 - Week 1

### PR DESCRIPTION
**Pull request by MarinaAlmeida20**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: 1-exercises/A-accessing-values/exercise1.js [hunk 1]
```diff
@@ -8,21 +8,32 @@ let dog = {
...
+for (let elem in dog) {
+  console.log(typeof dog[elem]);
+}
+/*
+output:
+string
+string
+boolean
+number
+*/
```
Comment (jackhclee) [inline 1-exercises/A-accessing-values/exercise1.js:23]: Great use of typeof operator!

File: 1-exercises/A-accessing-values/exercise2.js [hunk 2]
```diff
@@ -7,22 +7,32 @@
 let capitalCities = {
   UnitedKingdom: "London",
   China: "Beijing",
-  Peru: "Lima"
+  Peru: "Lima",
```
Comment (jackhclee) [inline 1-exercises/A-accessing-values/exercise2.js:10]: The last comma is optional

File: 1-exercises/A-accessing-values/exercise3.js [hunk 3]
```diff
@@ -20,13 +20,13 @@ let basketballTeam = {
     - console.logs the name of each player on a new line
 */
 
-// write code here
-
+const topPlayersArray = basketballTeam.topPlayers;
+topPlayersArray.sort().forEach((element) => console.log(element));
```
Comment (jackhclee) [inline 1-exercises/A-accessing-values/exercise3.js:24]: Good!

File: 1-exercises/B-setting-values/exercise1.js [hunk 4]
```diff
@@ -21,8 +21,13 @@ let capitalCities = {
...
     - Add a population of 9750000 to Peru's capital city.
 */
+capitalCities.UnitedKingdom.population = 8980000;
+capitalCities.China.population = 21500000;
 
-// write code here
+capitalCities.Peru = {
+  name: "Lima",
+  population: 9750000,
+}; // to create a new object inside an object
```
Comment (jackhclee) [inline 1-exercises/B-setting-values/exercise1.js:30]: Good

File: 1-exercises/B-setting-values/exercise2.js [hunk 5]
```diff
@@ -6,7 +6,7 @@
 let student = {
   name: "Reshma Saujani",
   examScore: 65,
-  hasPassed: false
+  hasPassed: false,
```
Comment (jackhclee) [inline 1-exercises/B-setting-values/exercise2.js:9]: The use of comma for last element is optional

File: 1-exercises/C-undefined-properties/exercise.js [hunk 6]
```diff
@@ -15,24 +15,27 @@ let car = {
...
+// this is undefined because the user is a parameter of the function and when the function calls on line 30 it is calling nothing
 
 let user = {
-  name: "Mira"
+  // the user here is an object
+  name: "Mira",
 };
 
-sayHelloToUser(user);
+sayHelloToUser(user); // it is calling just the Hello the other part has nothing
```
Comment (jackhclee) [inline 1-exercises/C-undefined-properties/exercise.js:31]: Correct

File: 1-exercises/C-undefined-properties/exercise.js [hunk 7]
```diff
@@ -15,24 +15,27 @@ let car = {
...
 let myPet = {
   animal: "Cat",
-  getName: function() {
+  getName: function () {
     "My pet's name is Fluffy";
   },
 };
 
-console.log(myPet.getName());
+console.log(myPet.getName()); // this function does't return nothing
```
Comment (jackhclee) [inline 1-exercises/C-undefined-properties/exercise.js:41]: Correct

File: 1-exercises/D-object-methods/exercise.js [hunk 8]
```diff
@@ -8,13 +8,15 @@
 */
 
 let student = {
-  // write code here
-}
+  getName: function (name) {
+    console.log(`Student name: ${name}`);
+  },
+};
```
Comment (jackhclee) [inline 1-exercises/D-object-methods/exercise.js:14]: Good

File: 2-mandatory/1-recipes.js [hunk 9]
```diff
@@ -22,4 +22,113 @@
...
+  Title: "Jelly",
+  Serves: 3,
+  Ingredients: ["hot water", "cold water", "jelly", "sugar"],
+};
+
+console.log(favoriteRecipe.Title);
+console.log(`Serves: ${favoriteRecipe.Serves}`);
+console.log("Ingredients:");
+let ingred = favoriteRecipe.Ingredients;
+ingred.forEach((e) => console.log(e));
```
Comment (jackhclee) [inline 2-mandatory/1-recipes.js]: Your submission fine. Please think of 2 ways to minimise code duplication?

File: 2-mandatory/2-currency-code-lookup.js [hunk 10]
```diff
@@ -17,9 +17,8 @@ const COUNTRY_CURRENCY_CODES = [
   ["MX", "MXN"],
 ];
 
-function createLookup(countryCurrencyCodes) {
-  // write code here
-}
+const createLookup = (countryCurrencyCodes) =>
+  Object.fromEntries(countryCurrencyCodes);
```
Comment (jackhclee) [inline 2-mandatory/2-currency-code-lookup.js:21]: Great use of built-in library!

File: 2-mandatory/3-shopping-list.js [hunk 11]
```diff
@@ -19,7 +19,19 @@ let pantry = {
...
+      return false; // if has the ingredient , remove it
+    } else {
+      return true; // if has not the ingredient, put it in the object
+    }
+  });
+  const objectReturn = {
+    name: recipe.name,
+    items: missingItems,
+  };
+  return objectReturn;
```
Comment (jackhclee) [inline 2-mandatory/3-shopping-list.js:34]: Good written function. Some advices on variable naming
1. allIngredients => ingredientsInPantry
2. "if has not" => if ingredient missing

File: 1-exercises/A-accessing-values/exercise2.js (hunk 2 above)
Comment (MarinaAlmeida20) [reply 1-exercises/A-accessing-values/exercise2.js:10]: It comes out after I save the code, maybe it is some VSCode extension.

File: 2-mandatory/1-recipes.js (hunk 9 above)
Comment (MarinaAlmeida20) [reply 2-mandatory/1-recipes.js]: I'll refactor this code, and submit it again later.

**Pull request closed without merging**